src/main/java/com/example/compound/cli_controllers/Controller.java

```diff
@@ -25,7 +27,7 @@
     public CurrentUserManager currentUserManager;
 
     public Controller(RepositoryGatewayI<BudgetTransferData> budgetRepository,
-                      RepositoryGatewayI<Group> groupRepository,
+                      GroupRepository groupRepository,
```

Comment:
GroupRepository "is a" RepositoryGatewayI<Group>, so a RepositoryGatewayI<Group> variable could be an example of the Open/Closed Principle.

I do sometimes get errors related to this line. I think they are a mistake because they go away when I restart IntelliJ, but what if the errors occur during marking?

Reply:
Chances are it won't since I doubt they will make changes on the file, since I think the error only occours during then